src/node/nodestate.h

```diff
@@ -603,6 +603,42 @@ namespace ccf
       }
     }
 
+    void node_quotes(Store::Tx& tx, nlohmann::json& j)
+    {
+      auto nodes_view = tx.get_view(network.nodes);
+
+      nlohmann::json quotes;
+      nodes_view->foreach([&quotes](const NodeId& nid, const NodeInfo& ni) {
+        if (ni.status == ccf::NodeStatus::TRUSTED)
+        {
+          nlohmann::json quote;
+          quote["raw"] = std::string(ni.quote.begin(), ni.quote.end());
+
+#ifdef GET_QUOTE
+          oe_report_t parsed_quote = {0};
+          auto res =
+            oe_parse_report(ni.quote.data(), ni.quote.size(), &parsed_quote);
+          if (res != OE_OK)
+          {
+            LOG_FAIL << "Failed to parse quote: " << oe_result_str(res)
```

Comment:
Should the RPC return an error message if the quote cannot be parsed?

Reply:
That's a good question, there are multiple quotes (one per node), so currently we return all the raw quotes, and the mrenclave for the ones we could parse. We could populate error messages for the rest?

I'm reluctant to make the whole RPC error because of this, it seems better to degrade gracefully and at least provide the raw quotes.

Reply:
Agreed. The `"mrenclave"` field could be "Quote could not be parsed" in that case.

Reply:
I'd rather not rely on field parsing/magic values and have something that looks self-explanatory in a schema, so I propose instead, on success:

{"raw": ..., "parsed": {"mrenclave": ...}}

on error:

{"raw": ..., "error": ...}